backend/app/tests/services/llm/test_multimodal.py

```
@@ -0,0 +1,450 @@
+    def test_valid_parts(self):
+        mm = MultiModalInput(
+            parts=[
+                TextContent(value="hello"),
+                ImageContent(format="base64", value="abc", mime_type="image/png"),
+                PDFContent(format="base64", value="abc", mime_type="application/pdf"),
+            ]
+        )
+        assert len(mm.parts) == 3
+
+    def test_empty_parts_raises(self):
+        with pytest.raises(Exception):
```

Ruff (0.15.2)

[warning] Do not assert blind exception: `Exception`

(B017)